### PR TITLE
Add CI to run maven tests

### PR DESCRIPTION
**Pull request by aforward**
Based on (if PR merged, then change to `main`)
https://github.com/eclipse-pass/main/blob/174-pipeline-documentation/docs/pipeline.md

Integrating the `mvn test` into GH actions.  Pinged @markpatton directly for some help deciphering the error messages.

### CODE REVIEW

File: .github/workflows/tests-unit.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        with:
+          java-version: 14
+      - name: Cache Maven packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Run unit tests
+        run: mvn -B test --file pom.xml
```
Comment (aforward) [inline .github/workflows/tests-unit.yml:22]: And thanks a ton @grant-mcs it was a _breeze_ to copy and paste from playground.

**Comment by aforward** [comment]
Did a temporary commit to test against this branch (that has the GH actions)

Being run here
https://github.com/eclipse-pass/pass-deposit-services/actions/runs/2339274756

And it failed :-(

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.18.1:test (default-test) on project shared-assembler: There are test failures.
[3024](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3024)
Error:  
[3025](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3025)
Error:  Please refer to /home/runner/work/pass-deposit-services/pass-deposit-services/shared-assembler/target/surefire-reports for the individual test results.
[3026](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3026)
Error:  -> [Help 1]
[3027](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3027)
Error:  
[3028](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3028)
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
[3029](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3029)
Error:  Re-run Maven using the -X switch to enable full debug logging.
[3030](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3030)
Error:  
[3031](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3031)
Error:  For more information about the errors and possible solutions, please read the following articles:
[3032](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3032)
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[3033](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3033)
Error:  
[3034](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3034)
Error:  After correcting the problems, you can resume the build with the command
[3035](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3035)
Error:    mvn <args> -rf :shared-assembler
[3036](https://github.com/eclipse-pass/pass-deposit-services/runs/6472536417?check_suite_focus=true#step:5:3036)
Error: Process completed with exit code 1.
```

**Comment by aforward** [comment]
Looks like mockito is upset

```
Underlying exception : org.mockito.exceptions.base.MockitoException: Could not modify all classes [class java.net.URL, class java.lang.Object, interface java.io.Serializable]
```

File: .github/workflows/tests-unit.yml (hunk 1 above)
Comment (aforward) [reply .github/workflows/tests-unit.yml:22]: @markpatton should this be `mvn test` and `mvn verify`?

Comment (aforward) [reply .github/workflows/tests-unit.yml:22]: We will capture `verify` separately as an integration test and not run as often as it's much slower

**Review by grant-cf — APPROVED** [review]

**Pull request merged**